core/src/main/java/com/oath/oak/ValueUtilsImpl.java

```diff
@@ -154,9 +161,9 @@ public ValueResult compute(Slice s, Consumer<OakWBuffer> computer, int version)
     }
 
     @Override
-    public <V> Result<V> exchange(Chunk<?, V> chunk, Chunk.LookUp lookUp, V value,
-                                  Function<ByteBuffer, V> valueDeserializeTransformer, OakSerializer<V> serializer,
-                                  MemoryManager memoryManager) {
+    public <V> Result<V> exchange(Chunk<?, V> chunk, EntrySet.LookUp lookUp, V value,
+        Function<ByteBuffer, V> valueDeserializeTransformer, OakSerializer<V> serializer,
+        MemoryManager memoryManager, InternalOakMap internalOakMap) {
```

Comment:
It can move to InternalOakMap, but then InternalOakMap will need to deal with locking/unlocking/changing off-heap header, which is unlikely... So it is doubtful, and can be done in other PR, we can speak about it. Resolving for now, reopen if you find it is needed.